src/psyclone/domain/common/transformations/alg_trans.py

```diff
@@ -88,9 +88,11 @@ def apply(self, psyir, options=None):
 
         '''
         self.validate(psyir, options=options)
+        idx = 0
         for call in psyir.walk(Call):
             if call.routine.name.lower() == "invoke":
-                self._invoke_trans.apply(call, options=options)
+                self._invoke_trans.apply(call, idx, options=options)
```

Comment:
I did not forsee that the index would be computed in the meta-transformation that calls the InvokeCallTrans, but this is right, it should be done here.

I am a bit uncomfortable breaking the transformation abstract apply() API with the additional argument but a grep shows that it in not the only transformation that does this. Maybe when we have the different raising->psyclone->lowering steps decided and how it works in the other APIs we can give this another look.

Reply:
I'm happy with that.